### PR TITLE
Infra proxy client details page UI

### PR DESCRIPTION
**Pull request by vinay033**
### :nut_and_bolt: Description: What code changed, and why?

Introduced client details page view.
### :chains: Related Resources
https://github.com/chef/automate/issues/3489
### :+1: Definition of Done
Client details page of infra view where we are displaying the client related details and public key tab.
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab

To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
Go to Infrastructure tab >> Chef Servers it's under Chef-Server feature flag.
Click on Server list >> Organisation list >> clients Tab >> then click on any of the client list items
```
### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots
client-details

### CODE REVIEW

File: components/automate-ui/src/app/entities/clients/client-details.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    case ClientActionTypes.GET:
+      return set(
+        GET_STATUS,
+        EntityStatus.loading,
+        clientDetailsEntityAdapter.removeAll(state)
+      ) as ClientDetailsEntityState;
+
+    case ClientActionTypes.GET_SUCCESS:
+      return set(GET_STATUS, EntityStatus.loadingSuccess,
+        clientDetailsEntityAdapter.addOne(action.payload, state));
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/entities/clients/client-details.reducer.ts:38]: does this return need the same treatment as above on line 34: ` as ClientDetailsEntityState;`?

File: components/automate-ui/src/app/entities/clients/client-details.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    case ClientActionTypes.GET_FAILURE:
+      return set(GET_STATUS, EntityStatus.loadingFailure, state);
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/entities/clients/client-details.reducer.ts:40]: same question as above does this return need the same treatment as above on line 34: as ClientDetailsEntityState;?

File: components/automate-ui/src/app/modules/infra-proxy/client-details/client-details.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        MockComponent({ selector: 'input', inputs: ['resetOrigin'] }),
+        MockComponent({ selector: 'chef-button',
+          inputs: ['disabled', 'routerLink'] }),
+        MockComponent({ selector: 'mat-select' }),
+        MockComponent({ selector: 'chef-th' }),
+        MockComponent({ selector: 'chef-td' }),
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/client-details/client-details.component.spec.ts:23]: duplicates to lines 34 and 35

File: components/automate-ui/src/app/entities/clients/client-details.reducer.ts (hunk 1 above)
Comment (vinay033) [reply components/automate-ui/src/app/entities/clients/client-details.reducer.ts]: I have added :)

**Pull request merged**